app/gosns/publish_batch.go

```diff
@@ -0,0 +1,111 @@
+package gosns
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/Admiral-Piett/goaws/app"
+	"github.com/Admiral-Piett/goaws/app/common"
+	"github.com/Admiral-Piett/goaws/app/interfaces"
+	"github.com/Admiral-Piett/goaws/app/models"
+	"github.com/Admiral-Piett/goaws/app/utils"
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+func PublishBatchV1(req *http.Request) (int, interfaces.AbstractResponseBody) {
+	requestBody := models.NewPublishBatchRequest()
+	ok := utils.REQUEST_TRANSFORMER(requestBody, req, false)
+	if !ok {
+		log.Error("Invalid Request - PublishBatchV1")
+		return utils.CreateErrorResponseV1("InvalidParameterValue", false)
+	}
+
+	if requestBody.TopicArn == "" {
+		return utils.CreateErrorResponseV1("InvalidParameterValue", false)
+	}
+
+	topicArn := requestBody.TopicArn
+	arnSegments := strings.Split(topicArn, ":")
+	topicName := arnSegments[len(arnSegments)-1]
+	topic, ok := app.SyncTopics.Topics[topicName]
+	if !ok {
+		return utils.CreateErrorResponseV1("TopicNotFound", false)
+	}
+
+	seen := make(map[string]bool)
+
+	for _, entry := range requestBody.PublishBatchRequestEntries {
```

Comment:
Can this whole loop be merged with the similar loop below?  Feels like we could just add the following condition below at the start of the second loop and be good?

```
if seen[entry.ID] {
	return utils.CreateErrorResponseV1("BatchEntryIdsNotDistinct", false)
}
```

Reply:
I don't think so because we will publish entries before realising there is a duplicate entry.ID when I believe we should return an error response and not publish any previous entities.